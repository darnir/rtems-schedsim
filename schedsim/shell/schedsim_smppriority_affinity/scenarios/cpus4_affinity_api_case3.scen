#
#  Verify that affinity APIs work with the right values
#
#  This scenario does **NOT** change the affinity of any thread
#  which ever executes.
#
rtems_init 4
task_create smp1 5
current_cpu 3

dispatch
cpus IDLE IDLE IDLE smp1

task_get_affinity smp1
task_set_affinity smp1 0x08
task_get_affinity smp1
cpus IDLE IDLE IDLE smp1

task_set_affinity smp1 0x04
current_cpu 2
task_get_affinity smp1
cpus IDLE IDLE smp1 IDLE

task_set_affinity smp1 0x02
current_cpu 1
task_get_affinity smp1
cpus IDLE smp1 IDLE IDLE

task_set_affinity smp1 0x01
current_cpu 0
task_get_affinity smp1
cpus smp1 IDLE IDLE IDLE
