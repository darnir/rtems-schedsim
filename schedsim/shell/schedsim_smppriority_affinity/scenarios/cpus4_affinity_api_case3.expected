Script File               : cpus4_affinity_api_case3.scen
verbose                   : 1
==> 1: #
==> 2: #  Verify that affinity APIs work with the right values
==> 3: #
==> 4: #  This scenario does **NOT** change the affinity of any thread
==> 5: #  which ever executes.
==> 6: #
==> 7: rtems_init 4
  CPU 0: Thread Heir: 0x09010001 priority=255
  CPU 0: Thread Executing: 0x09010001 priority=255
  CPU 1: Thread Heir: 0x09010002 priority=255
  CPU 1: Thread Executing: 0x09010002 priority=255
  CPU 2: Thread Heir: 0x09010003 priority=255
  CPU 2: Thread Executing: 0x09010003 priority=255
  CPU 3: Thread Heir: 0x09010004 priority=255
  CPU 3: Thread Executing: 0x09010004 priority=255
==> 8: task_create smp1 5
Creating task with:  Preemption: yes Timeslicing: no
Task (smp1) created: id=0x0a010001, priority=5
Task (smp1) starting: id=0x0a010001, priority=5
  CPU 3: Thread Heir: 0x0a010001 priority=5
  CPU 3: Thread Executing: 0x0a010001 priority=5
==> 9: current_cpu 3
Changing current CPU from 0 to 3
==> 10: 
==> 11: dispatch
==> 12: cpus IDLE IDLE IDLE smp1
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x09010001 @255 / 0x09010001 @255          false
  CPU 1: 0x09010002 @255 / 0x09010002 @255          false
  CPU 2: 0x09010003 @255 / 0x09010003 @255          false
  CPU 3: 0x0a010001 @  5 / 0x0a010001 @  5          false
=== End of Ready Set of Threads
==> 13: 
==> 14: task_get_affinity smp1
Task (0x0a010001) Get affinity=0x0000000f
==> 15: task_set_affinity smp1 0x08
Task (0x0a010001) Set affinity=0x00000008
==> 16: task_get_affinity smp1
Task (0x0a010001) Get affinity=0x00000008
==> 17: cpus IDLE IDLE IDLE smp1
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x09010001 @255 / 0x09010001 @255          false
  CPU 1: 0x09010002 @255 / 0x09010002 @255          false
  CPU 2: 0x09010003 @255 / 0x09010003 @255          false
  CPU 3: 0x0a010001 @  5 / 0x0a010001 @  5          false
=== End of Ready Set of Threads
==> 18: 
==> 19: task_set_affinity smp1 0x04
  CPU 2: Thread Heir: 0x0a010001 priority=5
  CPU 2: Thread Executing: 0x0a010001 priority=5
  CPU 3: Thread Heir: 0x09010004 priority=255
  CPU 3: Thread Executing: 0x09010004 priority=255
Task (0x0a010001) Set affinity=0x00000004
==> 20: current_cpu 2
Changing current CPU from 3 to 2
==> 21: task_get_affinity smp1
Task (0x0a010001) Get affinity=0x00000004
==> 22: cpus IDLE IDLE smp1 IDLE
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x09010001 @255 / 0x09010001 @255          false
  CPU 1: 0x09010002 @255 / 0x09010002 @255          false
  CPU 2: 0x0a010001 @  5 / 0x0a010001 @  5          false
  CPU 3: 0x09010004 @255 / 0x09010004 @255          false
=== End of Ready Set of Threads
==> 23: 
==> 24: task_set_affinity smp1 0x02
  CPU 1: Thread Heir: 0x0a010001 priority=5
  CPU 1: Thread Executing: 0x0a010001 priority=5
  CPU 2: Thread Heir: 0x09010003 priority=255
  CPU 2: Thread Executing: 0x09010003 priority=255
Task (0x0a010001) Set affinity=0x00000002
==> 25: current_cpu 1
Changing current CPU from 2 to 1
==> 26: task_get_affinity smp1
Task (0x0a010001) Get affinity=0x00000002
==> 27: cpus IDLE smp1 IDLE IDLE
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x09010001 @255 / 0x09010001 @255          false
  CPU 1: 0x0a010001 @  5 / 0x0a010001 @  5          false
  CPU 2: 0x09010003 @255 / 0x09010003 @255          false
  CPU 3: 0x09010004 @255 / 0x09010004 @255          false
=== End of Ready Set of Threads
==> 28: 
==> 29: task_set_affinity smp1 0x01
  CPU 0: Thread Heir: 0x0a010001 priority=5
  CPU 0: Thread Executing: 0x0a010001 priority=5
  CPU 1: Thread Heir: 0x09010002 priority=255
  CPU 1: Thread Executing: 0x09010002 priority=255
Task (0x0a010001) Set affinity=0x00000001
==> 30: current_cpu 0
Changing current CPU from 1 to 0
==> 31: task_get_affinity smp1
Task (0x0a010001) Get affinity=0x00000001
==> 32: cpus smp1 IDLE IDLE IDLE
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010001 @  5 / 0x0a010001 @  5          false
  CPU 1: 0x09010002 @255 / 0x09010002 @255          false
  CPU 2: 0x09010003 @255 / 0x09010003 @255          false
  CPU 3: 0x09010004 @255 / 0x09010004 @255          false
=== End of Ready Set of Threads
